File: 0011.ContainerWithMostWater/ContainerWithMostWater.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        llen = len(height)
        ret = 0
        i,j = 0,llen - 1
        while (i < j):
            if height[i] <= height[j]:
                ret = max(ret, (j - i) * height[i])
                i += 1
            else:
                ret = max(ret, (j - i) * height[j])
                j -= 1
        return ret

File: 0011.ContainerWithMostWater/ContainerWithMostWater.c
int maxArea(int* height, int heightSize)
{
    int ret = 0, k;
    int i = 0, j = heightSize - 1;
    while (i < j) {
        if (height[i] < height[j]) {
            k = height[i] * (j - i);
            i++;
        } else if (height[i] > height[j]) {
            k = height[j] * (j - i);
            j--;            
        } else {
            k = height[j] * (j - i);
            j--,i++;            
        }
        ret = k > ret? k : ret;
    }
    return ret;
}